plugin.video.embycon/resources/lib/functions.py

```diff
@@ -0,0 +1,1333 @@
+        tabFileName = __addondir__ + "profile(" + fileTimeStamp + ").txt"
+        f = open(tabFileName, 'wb')
+
+        s = StringIO.StringIO()
+        ps = pstats.Stats(pr, stream=s)
+        ps = ps.sort_stats('cumulative')
+        ps.print_stats()
+        ps.strip_dirs()
+        ps = ps.sort_stats('tottime')
+        ps.print_stats()
+        f.write(s.getvalue())
+
+        '''
+        ps = pstats.Stats(pr)
+        f.write("NumbCalls\tTotalTime\tCumulativeTime\tFunctionName\tFileName\r\n")
+        for (key, value) in ps.stats.items():
+            (filename, count, func_name) = key
+            (ccalls, ncalls, total_time, cumulative_time, callers) = value
+            try:
+                f.write(str(ncalls) + "\t" + "{:10.4f}".format(total_time) + "\t" + "{:10.4f}".format(cumulative_time) + "\t" + func_name + "\t" + filename + "\r\n")
+            except ValueError:
+                f.write(str(ncalls) + "\t" + "{0}".format(total_time) + "\t" + "{0}".format(cumulative_time) + "\t" + func_name + "\t" + filename + "\r\n")
+        '''
+
+        f.close()
+
+    log.debug("===== EmbyCon FINISHED =====")
+
+
+def markWatched(item_id):
+    log.debug("Mark Item Watched : " + item_id)
+    url = "{server}/emby/Users/{userid}/PlayedItems/" + item_id
+    downloadUtils.downloadUrl(url, postBody="", method="POST")
+    home_window = HomeWindow()
+    home_window.setProperty("force_data_reload", "true")
+    xbmc.executebuiltin("Container.Refresh")
+
+
+def markUnwatched(item_id):
+    log.debug("Mark Item UnWatched : " + item_id)
+    url = "{server}/emby/Users/{userid}/PlayedItems/" + item_id
+    downloadUtils.downloadUrl(url, method="DELETE")
+    home_window = HomeWindow()
+    home_window.setProperty("force_data_reload", "true")
+    xbmc.executebuiltin("Container.Refresh")
+
+
+def markFavorite(item_id):
+    log.debug("Add item to favourites : " + item_id)
+    url = "{server}/emby/Users/{userid}/FavoriteItems/" + item_id
+    downloadUtils.downloadUrl(url, postBody="", method="POST")
+    home_window = HomeWindow()
+    home_window.setProperty("force_data_reload", "true")
+    xbmc.executebuiltin("Container.Refresh")
+
+
+def unmarkFavorite(item_id):
+    log.debug("Remove item from favourites : " + item_id)
+    url = "{server}/emby/Users/{userid}/FavoriteItems/" + item_id
+    downloadUtils.downloadUrl(url, method="DELETE")
+    home_window = HomeWindow()
+    home_window.setProperty("force_data_reload", "true")
+    xbmc.executebuiltin("Container.Refresh")
+
+
+def delete(item_id):
+    return_value = xbmcgui.Dialog().yesno(i18n('confirm_file_delete'), i18n('file_delete_confirm'))
+    if return_value:
+        log.debug('Deleting Item : ' + item_id)
+        url = '{server}/emby/Items/' + item_id
+        progress = xbmcgui.DialogProgress()
+        progress.create(i18n('deleting'), i18n('waiting_server_delete'))
+        downloadUtils.downloadUrl(url, method="DELETE")
+        progress.close()
```

Comment:
You never update the percentage of this progressdialog. It might be too fast you won't notice that though.